python/cugraph-nx/pyproject.toml

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+[build-system]
+
+requires = [
+    "setuptools",
+    "wheel",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "cugraph-nx"
+version = "23.10.00"
+description = "cugraph backend for NetworkX"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "NVIDIA Corporation" },
+]
+license = { text = "Apache 2.0" }
+requires-python = ">=3.9"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3 :: Only",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = [
+    "cupy-cuda11x>=12.0.0",
+    "networkx>=3.0",
+    "pylibcugraph==23.10.*",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+
+[project.optional-dependencies]
+test = [
+    "networkx>=3.0",
+    "numpy>=1.21",
+    "pandas",
+    "pytest",
+    "pytest-benchmark",
+    "pytest-cov",
+    "pytest-mpl",
+    "pytest-xdist",
+    "scipy",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
```

Comment:
Thanks for filling this out.

I did add coverage to `run_nx_tests.sh`, so we do need `pytest-cov`. `pytest-xdist` is _sometimes_ useful (and sometimes it uses all GPU memory).

We don't yet use `pandas` anywhere, and `scipy` is only used in a benchmark. But, these are in the "default" networkx dependencies, which we probably want to include if we plan to run networkx tests: https://github.com/networkx/networkx/blob/main/requirements/default.txt

btw, `pytest-mpl` will install `matplotlib`, which was included in "default.txt".

Reply:
I think this is coming from the addition of `test_python_common` to cugraph-nx in `dependencies.yaml`, but perhaps I should redefine that for a new definition of "common" now.  I'll take care of this in the next commit.

Reply:
This still isn't completely resolved in the latest commit.  I still need to add the [networkx test deps](https://github.com/networkx/networkx/blob/main/requirements/default.txt) either as a test dep in depedencies.yaml, or in our CI script when we run against networkx tests.

Reply:
We'll probably want to add `scipy` and `pandas` to our test dependencies soon, so I'm okay if we have them listed here (or we could wait until we actually need them). Having them will help run some networkx tests.